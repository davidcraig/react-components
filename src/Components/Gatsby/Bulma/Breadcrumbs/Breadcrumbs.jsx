import React from 'react'
import { Link } from 'gatsby'
import PropTypes from 'prop-types'

export function ActiveBreadcrumb (props) {
  return <li className='is-active' key={`${props.name}-breadcrumb`}><a href='#' aria-current='page'>{props.name}</a></li>
}
ActiveBreadcrumb.propTypes = {
  name: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired
}

/**
 * Breadcrumb (Gatsby + Bulma)
 * @param {*} props
 */
export function Breadcrumb (props) {
  return <li key={`${props.title}-breadcrumb`}><Link to={props.href}><a>{props.title}</a></Link></li>
}
Breadcrumb.propTypes = {
  title: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired
}

/**
 * Breadcrumbs (Gatsby + Bulma)
 * @param {*} props
 */
export function Breadcrumbs (props) {
  return <nav className='breadcrumb has-succeeds-separator' aria-label='breadcrumbs'>
    <ul>{props.children}</ul>
  </nav>
}
Breadcrumbs.propTypes = {
  children: PropTypes.any
}
Breadcrumbs.defaultProps = {
  class: 'has-succeeds-seperator'
}
