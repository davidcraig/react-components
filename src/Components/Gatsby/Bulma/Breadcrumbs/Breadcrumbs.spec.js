// Link.react.test.js
import React from 'react'
import { Breadcrumb } from './Breadcrumbs'
import renderer from 'react-test-renderer'

beforeEach(() => {
  global.___loader = {
    enqueue: jest.fn()
  }
})

test('Breadcrumb renders as expected', () => {
  const component = renderer.create(
    <Breadcrumb title='Test Breadcrumb' href='/'></Breadcrumb>
  )

  expect(component.toJSON()).toMatchSnapshot()
})
