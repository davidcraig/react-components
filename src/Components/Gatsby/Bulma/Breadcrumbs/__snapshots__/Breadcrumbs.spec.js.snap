// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders as expected 1`] = `
<li>
  <a
    aria-current="page"
    className=""
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
  >
    <a>
      Test Breadcrumb
    </a>
  </a>
</li>
`;
